SU2_CFD/src/solver_structure.cpp

```diff
@@ -4825,6 +4825,365 @@ void CSolver::Read_InletFile_ASCII(CGeometry *geometry, CConfig *config, string
+
+        Marker_Tag = config->GetMarker_All_TagBound(iMarker);
+
+        for (jMarker = 0; jMarker < nMarker_InletFile; jMarker++) {
+
+          /*--- If we have found the matching marker string, continue. ---*/
+
+          if (Marker_Tags_InletFile[jMarker] == Marker_Tag) {
+
+            /*--- Increment our counter for marker matches. ---*/
+
+            Marker_Counter++;
+            
+            nVertex = geometry[MESH_0]->nVertex[iMarker];
+
+            /*--- Initializing a new container for inlet data for printing if INLET_DATA_OUT is true. ---*/
+            Inlet_InterpolatedData = new passivedouble[nVertex*(maxCol_InletFile+nDim)];
+
+            /*--- Loop through the nodes on this marker. ---*/
+
+            for (iVertex = 0; iVertex < geometry[MESH_0]->nVertex[iMarker]; iVertex++) {
+
+              iPoint   = geometry[MESH_0]->vertex[iMarker][iVertex]->GetNode();
+              Coord    = geometry[MESH_0]->node[iPoint]->GetCoord();
+
+              /*--- Finding radius and theta for the specific vertex. ---*/ 
+              Interp_Radius = sqrt(pow(Coord[0],2)+ pow(Coord[1],2));
+              Theta = atan2(Coord[1],Coord[0]);
+
+              /*--- Set the x,y,z coordinates in Inlet_Values container ---*/
+              for  (iVar=0; iVar < nDim; iVar++)
+                Inlet_Values[iVar]=Coord[iVar];
+              
+              /*--- Loop through all rows in the inlet file. ---*/
+              for (iRow = nRowCum_InletFile[jMarker]; iRow < nRowCum_InletFile[jMarker+1]; iRow++) {
+                   
+                   /*--- Find the two closest radii for the specific vertex. ---*/
+                  if (Inlet_Data[maxCol_InletFile*iRow] <= Interp_Radius && Inlet_Data[maxCol_InletFile*(iRow+1)] >= Interp_Radius){
+                      
+                      /*--- Loop through all values in the inlet file except the radii---*/
+                      for (unsigned short index=1; index<maxCol_InletFile; index++){
+                        Point_Match = true;
+                        switch(config->GetKindInletInterpolationFunction()){
+                          case (ONED_LINEAR_SPANWISE):
+                          /*--- 1D interpolate data for every index (column) using equation of a line. ---*/
+                          slope=(Inlet_Data[index+maxCol_InletFile*(iRow+1)]-Inlet_Data[index+maxCol_InletFile*iRow])/(Inlet_Data[maxCol_InletFile*(iRow+1)]-Inlet_Data[maxCol_InletFile*iRow]);
+                      
+                          /*--- If interpolating turbulence variables, shift them one index ahead ---*/
+                          if (index > nDim+1)
+                            Inlet_Values[index+(nDim-1)+1]=slope*(Interp_Radius - Inlet_Data[maxCol_InletFile*iRow]) + Inlet_Data[index+maxCol_InletFile*iRow];
+                          else
+                            Inlet_Values[index+(nDim-1)]=slope*(Interp_Radius - Inlet_Data[maxCol_InletFile*iRow]) + Inlet_Data[index+maxCol_InletFile*iRow];
```

Comment:
In situations like these it is better to handle just the index in the if statement to avoid repeating the entire line of code.
Also (micro optimization tip) if this is just to handle an index bound consider using `min/max` or transforming the conditional into arithmetic to avoid branching, i.e.:
```
offset = index > nDim+1;
Inlet_Values[index+nDim-1+offset] = ...
```

Reply:
I get your idea. Will look into it.